torch/distributed/_tensor/dispatch.py

```diff
@@ -174,6 +272,19 @@ def operator_dispatch(
         local_tensor_args = cast(Tuple[object, ...], local_tensor_args)
         local_tensor_kwargs = cast(Dict[str, object], local_tensor_kwargs)
         local_results = op_call(*local_tensor_args, **local_tensor_kwargs)
+        if (
+            (mesh is not None)
+            and (mesh.mesh.numel() < dist.get_world_size())
+        ):
+            # communicate the result to non-participating ranks if
+            # op runs on a submesh and return type is scalar value
+            ret_list = op_schema.func_schema.returns
+            if (
+                (len(ret_list) == 1)
+                and (_arg_type_is_primitive(str(ret_list[0].type)))
```

Comment:
I think `ret_list` is always length == 1, as function schema always have a single return (if it;s multiple return, it would return Tuple or List), are you saying for `List[Tensor]` case, it would actually have multiple return types?

Reply:
My confusion was that, in `native_functions.yaml` there're many functions that have multiple return values (e.g. search for `(Tensor, Tensor)`) but they seem to not hit the dispatcher at all. For example, aten functions like `native_dropout` return a tuple of tensors but it doesn't hit the dispatcher because it's `aten::native`. Is that correct?

Reply:
No, `native_dropout` did hit the dispatcher, if there's a `dispatch` key in the schema, it would hit the dispatcher

Reply:
I see. I wanted to see if for native_dropout (https://github.com/pytorch/pytorch/blob/cc699c56dc14af5b652a49e3044c0fe7c8ca70ee/aten/src/ATen/native/native_functions.yaml#L241-L248) whose return type is "(Tensor, Tensor)" the `ret_list` is still of length 1. So I tried:
```
dropout = torch.nn.Dropout(p=0.5)
dtensor = dropout(dtensor)
```
but the operator hitting dispatcher is actually `dropout` (https://github.com/pytorch/pytorch/blob/cc699c56dc14af5b652a49e3044c0fe7c8ca70ee/aten/src/ATen/native/native_functions.yaml#L269-L270). 

Do you know how to test with `native_dropout` function? Or, I misunderstood here. `native_dropout` is an ATen op which is called in the torch op `dropout`. Only `dropout` hits DTensor's dispatcher and `native_dropout` hits PyTorch dispatcher, is that right?

Reply:
there's no difference between DTensor dispatcher and pytorch's one. `native_dropout` is an aten op and we are only targetting on aten op, I think we registered `native_dropout` https://github.com/pytorch/pytorch/blob/60e503d4685f7cb4e7b7af5b11548c1d47d299e0/torch/distributed/_tensor/ops/pointwise_ops.py#L403 So don't consider torch op

If you don't see it, it's likely you are running on CPU instead of GPU, I remember somewhat native_dropout only get hit on GPU.